cmd/nerdctl/container_kill_linux_test.go

```diff
@@ -0,0 +1,91 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+	"testing"
+
+	"github.com/containerd/nerdctl/v2/pkg/rootlessutil"
+	"github.com/containerd/nerdctl/v2/pkg/testutil"
+	"github.com/coreos/go-iptables/iptables"
+	"gotest.tools/v3/assert"
+)
+
+// TestKillCleanupForwards runs a container that exposes a port and then kill it.
+// The test checks that the kill command effectively clean up
+// the iptables forwards creted from the run.
+func TestKillCleanupForwards(t *testing.T) {
+	const (
+		hostPort          = 9999
+		testContainerName = "ngx"
+	)
+	base := testutil.NewBase(t)
+	defer func() {
+		base.Cmd("rm", "-f", testContainerName).Run()
+	}()
+
+	// skip if rootless
+	if rootlessutil.IsRootless() {
+		t.Skip("Nydusify check is not supported rootless mode.")
+	}
+
+	ipt, err := iptables.New()
+	assert.NilError(t, err)
+
+	cmd := base.Cmd("run", "-d",
+		"--restart=no",
+		"--name", testContainerName,
+		"-p", fmt.Sprintf("127.0.0.1:%d:80", hostPort),
+		testutil.NginxAlpineImage)
+	containerID := strings.TrimSpace(cmd.Run().Stdout())
+	assert.Equal(t, forwardExists(t, ipt, testutil.Namespace, containerID), true)
+
+	base.Cmd("kill", testContainerName).AssertOK()
+	assert.Equal(t, forwardExists(t, ipt, testutil.Namespace, containerID), false)
+}
+
+// forwardExists check that at least 2 rules are present in the CNI-HOSTPORT-DNAT chain
+// and checks for regex matches in the list of rules
+func forwardExists(t *testing.T, ipt *iptables.IPTables, namespace, containerID string) bool {
```

Comment:
This should be in a new package like `pkg/testutil/iptables`

Reply:
ok, I got your point, will provide it soon